File: BSvZP-Common/MessagesTester/JoinGameTester.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using Common;
using Messages;

namespace MessagesTester
{
    /// <summary>
    /// Summary description for JoinGame
    /// </summary>
    [TestClass]
    public class JoinGameTester
    {
        [TestMethod]
        public void JoinGame_TestConstructorsAndFactories()
        {
            JoinGame jg = new JoinGame();
            Assert.AreEqual(0, jg.GameId);
            Assert.IsNull(jg.AgentInfo);

            AgentInfo agentInfo = new AgentInfo(1001, AgentInfo.PossibleAgentType.BrilliantStudent) { ANumber = "A0001", FirstName = "Joe", LastName = "Jone" };
            jg = new JoinGame(10, agentInfo);
            Assert.AreEqual(10, jg.GameId);

            Assert.AreSame(agentInfo, jg.AgentInfo);

            ByteList bytes = new ByteList();
            jg.Encode(bytes);
            Message msg = Message.Create(bytes);
            Assert.IsNotNull(msg);
            Assert.IsTrue(msg is JoinGame);
            JoinGame jg2 = msg as JoinGame;
            Assert.AreEqual(jg.GameId, jg2.GameId);
        }

        [TestMethod]
        public void JoinGame_Properties()
        {
            AgentInfo agentInfo = new AgentInfo(1001, AgentInfo.PossibleAgentType.BrilliantStudent) { ANumber = "A00001", FirstName = "Joe", LastName = "Jones" };
            JoinGame jg = new JoinGame(10, agentInfo);
            Assert.AreEqual(10, jg.GameId);
            Assert.AreEqual("A00001", jg.AgentInfo.ANumber);
            Assert.AreEqual("Joe", jg.AgentInfo.FirstName);
            Assert.AreEqual("Jones", jg.AgentInfo.LastName);
            Assert.AreSame(agentInfo, jg.AgentInfo);

            jg.GameId = 20;
            Assert.AreEqual(20, jg.GameId);
            jg.AgentInfo = null;
            Assert.IsNull(jg.AgentInfo);
            jg.AgentInfo = agentInfo;
            Assert.AreSame(agentInfo, jg.AgentInfo);

            Assert.AreEqual(Message.MESSAGE_CLASS_IDS.JoinGame, jg.MessageTypeId());
        }

        [TestMethod]
        public void JoinGame_EncodingAndDecoding()
        {
            AgentInfo agentInfo = new AgentInfo(1001, AgentInfo.PossibleAgentType.BrilliantStudent) { ANumber = "A0001", FirstName = "Joe", LastName = "Jone" };
            JoinGame jg1 = new JoinGame(10, agentInfo);
            Assert.AreEqual(10, jg1.GameId);
            Assert.AreSame(agentInfo, jg1.AgentInfo);

            ByteList bytes = new ByteList();
            jg1.Encode(bytes);
            JoinGame jg2 = JoinGame.Create(bytes);
            Assert.AreEqual(jg1.GameId, jg2.GameId);


            bytes.Clear();
            jg1.Encode(bytes);
            bytes.GetByte();            // Read one byte, which will throw the length off
            try
            {
                jg2 = JoinGame.Create(bytes);
                Assert.Fail("Expected an exception to be thrown");
            }
            catch (ApplicationException)
            {
            }

            bytes.Clear();
            jg1.Encode(bytes);
            bytes.Add((byte)100);       // Add a byte
            bytes.GetByte();            // Read one byte, which will make the ID wrong
            try
            {
                jg2 = JoinGame.Create(bytes);
                Assert.Fail("Expected an exception to be thrown");
            }
            catch (ApplicationException)
            {
            }

        }

    }
}

File: BSvsZP-Common -Java/MessagesTester/JoinGameTester.java
package MessagesTester;

import static org.junit.Assert.*;

import org.junit.Test;
import org.omg.CORBA.portable.ApplicationException;

import Common.AgentInfo;
import Common.AgentInfo.PossibleAgentStatus;
import Common.ByteList;
import Common.ComponentInfo;
import Messages.JoinGame;
import Messages.Message;

public class JoinGameTester {

	@Test
	 public void JoinGame_TestConstructorsAndFactories() throws Exception
    {
        JoinGame jg = new JoinGame();
        assertEquals(0, jg.getGameId());
        assertNull(jg.getAgentInfo());

        AgentInfo agentInfo = new AgentInfo((short) 1001, AgentInfo.PossibleAgentType.BrilliantStudent);
        agentInfo.setFirstName("Joe");
        agentInfo.setLastName("Jones");
        agentInfo.setANumber("A00123");
        agentInfo.setAgentStatus(PossibleAgentStatus.InGame);
        
        jg = new JoinGame((short) 10, agentInfo);
        
        assertEquals(10, jg.getGameId());
        assertSame(agentInfo, jg.getAgentInfo());
        assertSame(agentInfo, jg.getAgentInfo());

        ByteList bytes = new ByteList();
        
        jg.Encode(bytes);
        
        Message msg = Message.Create(bytes);
        
        assertNotNull(msg);
        
        
        JoinGame jg2 = (JoinGame) msg ;
        
        assertEquals(jg.getGameId(), jg2.getGameId());
    }

	@Test
	public void JoinGame_Properties()
    {
		AgentInfo agentInfo = new AgentInfo((short) 1001, AgentInfo.PossibleAgentType.BrilliantStudent);
		agentInfo.setFirstName("Joe");
	    agentInfo.setLastName("Jones");
	    agentInfo.setANumber("A00123");
	    agentInfo.setAgentStatus(PossibleAgentStatus.InGame);
	    
        JoinGame jg = new JoinGame((short)10, agentInfo);
        
        assertEquals(10, jg.getGameId());
        assertEquals("A00123", jg.getAgentInfo().getANumber());
        assertEquals("Jones", jg.getAgentInfo().getLastName());
        assertEquals("Joe", jg.getAgentInfo().getFirstName());
        assertSame(agentInfo, jg.getAgentInfo());

        jg.setGameId((short) 20);
        assertEquals(20, jg.getGameId());

        jg.setAgentInfo(null);
        assertNull(jg.getAgentInfo());
        
        jg.setAgentInfo(agentInfo);
        assertSame(agentInfo, jg.getAgentInfo());
        
        
        assertEquals(Message.MESSAGE_CLASS_IDS.JoinGame.getValue(), jg.MessageTypeId().getValue());
        assertEquals(Message.MESSAGE_CLASS_IDS.JoinGame, jg.MessageTypeId());
    }
   
	@Test
	 public void JoinGame_EncodingAndDecoding() throws ApplicationException, Exception
    {
        AgentInfo agentInfo = new AgentInfo((short) 1001, AgentInfo.PossibleAgentType.BrilliantStudent);
        agentInfo.setFirstName("Joe");
        agentInfo.setLastName("Jones");
        agentInfo.setANumber("A00123");
        agentInfo.setAgentStatus(PossibleAgentStatus.InGame);
        
        JoinGame jg1 = new JoinGame((short) 10, agentInfo);
        
        assertEquals(10, jg1.getGameId());
        assertSame(agentInfo, jg1.getAgentInfo());
        
        ByteList bytes = new ByteList();
       
        jg1.Encode(bytes);
    
    	
		JoinGame jg2 = JoinGame.Create(bytes);
		
		        		
        assertEquals(jg1.getGameId(), jg2.getGameId());
      
        bytes.Clear();
        jg1.Encode(bytes);
        bytes.GetByte();            // Read one byte, which will throw the length off
        try
        {
            jg2 = JoinGame.Create(bytes);
            fail("Expected an exception to be thrown");
        }
        catch (ApplicationException e)
        {
        }

        bytes.Clear();
        jg1.Encode(bytes);
        bytes.Add((byte)100);       // Add a byte
        bytes.GetByte();            // Read one byte, which will make the ID wrong
        try
        {
            jg2 = JoinGame.Create(bytes);
            fail("Expected an exception to be thrown");
        }
        catch (ApplicationException e)
        {
        }
    }
	
	@Test
	public void JoinGame_EncodingAndDecodingTester() throws ApplicationException, Exception
	{
	     AgentInfo agentInfo = new AgentInfo((short) 1001, AgentInfo.PossibleAgentType.BrilliantStudent);
	     agentInfo.setFirstName("Joe");
	     agentInfo.setLastName("Jones");
	     agentInfo.setANumber("A00001");
	     agentInfo.setAgentStatus(AgentInfo.PossibleAgentStatus.InGame);
	        
	     assertEquals(1001, agentInfo.getId());
	     assertEquals(AgentInfo.PossibleAgentType.BrilliantStudent, agentInfo.getAgentType());
	     assertEquals("Joe", agentInfo.getFirstName());
	     assertEquals("Jones", agentInfo.getLastName());
	     assertEquals("A00001", agentInfo.getANumber());
	     assertEquals(AgentInfo.PossibleAgentStatus.InGame, agentInfo.getAgentStatus());
	     
	     
	     JoinGame jg = new JoinGame((short) 10, agentInfo);
	     ByteList bytes = new ByteList();
	        
	     jg.Encode(bytes);
	     assertEquals(10, jg.getGameId());
	     assertEquals("Joe", jg.getAgentInfo().getFirstName());
	     assertEquals("Jones", jg.getAgentInfo().getLastName());
	     assertEquals("A00001", jg.getAgentInfo().getANumber());
	     assertEquals(AgentInfo.PossibleAgentStatus.InGame, jg.getAgentInfo().getAgentStatus());
	     
	     
	     byte[] b = bytes.GetBytes(bytes.getLength());
	     for(int i =0; i<b.length; ++i)
	    	 System.out.println(b[i]);
	 }
	
}
